client/src/components/TrackItem.vue

```diff
@@ -146,7 +146,21 @@ export default defineComponent({
           frame: frameRef.value,
           keyframe: true,
         });
-      } else {
+      } else if ((f.lower || f.upper) && !f.isKeyframe) {
+        let interFeature: Feature | null = null;
+        if (f.upper && frameRef.value > f.upper.frame) {
+          interFeature = f.upper;
+        } else if (f.lower && frameRef.value < f.lower.frame) {
+          interFeature = f.lower;
+        }
+        if (interFeature) {
+          props.track.setFeature({
+            ...interFeature,
+            frame: frameRef.value,
+            keyframe: true,
+          });
+        }
+      } else if (f.isKeyframe) {
```

Comment:
Would it be unreasonable to ask that `toggleKeyframe(frame: number)` be made an instance method in `track.ts`?

It's much easier to test there than it would be here.  You don't need to add tests.

Reply:
That's a great idea.  I'll do that and I'll add tests.  It feels kind of weird having that kind of logic/function live in a Vue component.  I think I still need to keep the computed `feature` for some stuff in the template though.